cumulusci/tasks/bulkdata.py

```diff
@@ -501,12 +582,54 @@ def _create_table(self, mapping):
         else:
             self.models[mapping["table"]] = type(model_name, (object,), {})
 
+        id_column = mapping["fields"].get("Id") or "id"
         fields = []
-        fields.append(Column("id", Integer, primary_key=True))
-        if "record_type" in mapping:
-            fields.append(Column("record_type", Unicode(255)))
+        fields.append(Column(id_column, Unicode(255), primary_key=True))
         for field in self._fields_for_mapping(mapping):
+            if field["sf"] == "Id":
+                continue
             fields.append(Column(field["db"], Unicode(255)))
+        if "record_type" in mapping:
+            fields.append(Column("record_type", Unicode(255)))
         t = Table(mapping["table"], self.metadata, *fields, **table_kwargs)
 
         mapper(self.models[mapping["table"]], t, **mapper_kwargs)
+
+    def _fields_for_mapping(self, mapping):
+        fields = []
+        for sf_field, db_field in mapping.get("fields", {}).items():
+            fields.append({"sf": sf_field, "db": db_field})
+        for sf_field, lookup in mapping.get("lookups", {}).items():
+            fields.append({"sf": sf_field, "db": lookup["key_field"]})
+        return fields
+
+
+@contextmanager
+def _download_file(uri, bulk_api):
+    """Download the bulk API result file for a single batch"""
+    resp = requests.get(uri, headers=bulk_api.headers(), stream=True)
+    with tempfile.TemporaryFile("w+b") as f:
+        for chunk in resp.iter_content(chunk_size=None):
+            f.write(chunk)
+        f.seek(0)
+        yield f
+
+
+def process_incoming_rows(f, record_type=None):
+    if record_type and not isinstance(record_type, bytes):
+        record_type = record_type.encode("utf-8")
+    for line in f:
+        if record_type:
+            yield line + b"," + record_type + b"\n"
+        else:
+            yield line
+
+
+def log_progress(iterable, logger, batch_size=10000):
```

Comment:
i think there miiight be a click helper for this?

Reply:
I don't really want task code to depend on click though.

Reply:
throw this in tasks.utils then